aptos-move/framework/aptos-stdlib/sources/cryptography/pedersen.move

```diff
@@ -0,0 +1,124 @@
+        let rand_base = std::option::extract(&mut rand_base);
+
+        Commitment {
+            point: ristretto255::basepoint_double_mul(rand, &rand_base, val)
+        }
+    }
+
+    /// Returns a non-hiding commitment val * basepoint where `basepoint` is the Ristretto255 basepoint.
+    public fun new_non_hiding_commitment_for_bulletproof(val: &Scalar): Commitment {
+        Commitment {
+            point: ristretto255::basepoint_mul(val)
+        }
+    }
+
+    /// Returns lhs + rhs. Useful for re-randomizing the commitment or updating the committed value.
+    public fun commitment_add(lhs: &Commitment, rhs: &Commitment): Commitment {
+        Commitment {
+            point: ristretto255::point_add(&lhs.point, &rhs.point)
+        }
+    }
+
+    /// Sets lhs = lhs + rhs. Useful for re-randomizing the commitment or updating the committed value.
+    public fun commitment_add_assign(lhs: &mut Commitment, rhs: &Commitment) {
+        ristretto255::point_add_assign(&mut lhs.point, &rhs.point);
+    }
+
+    /// Returns lhs - rhs. Useful for re-randomizing the commitment or updating the committed value.
+    public fun commitment_sub(lhs: &Commitment, rhs: &Commitment): Commitment {
+        Commitment {
+            point: ristretto255::point_sub(&lhs.point, &rhs.point)
+        }
+    }
+
+    /// Sets lhs = lhs - rhs. Useful for re-randomizing the commitment or updating the committed value.
+    public fun commitment_sub_assign(lhs: &mut Commitment, rhs: &Commitment) {
+        ristretto255::point_sub_assign(&mut lhs.point, &rhs.point);
+    }
+
+    /// Creates a copy of this commitment.
+    public fun commitment_clone(c: &Commitment): Commitment {
+        Commitment {
+            point: ristretto255::point_clone(&c.point)
+        }
+    }
+
+    /// Returns true if the two commitments are identical: i.e., same value and same randomness.
+    public fun commitment_equals(lhs: &Commitment, rhs: &Commitment): bool {
+        ristretto255::point_equals(&lhs.point, &rhs.point)
+    }
+
+    /// Returns the underlying elliptic curve point representing the commitment as an in-memory RistrettoPoint.
+    public fun commitment_as_point(c: &Commitment): &RistrettoPoint {
+        &c.point
+    }
+
+    /// Returns the Pedersen commitment as a CompressedRistretto point.
+    public fun commitment_as_compressed_point(c: &Commitment): CompressedRistretto {
+        point_compress(&c.point)
+    }
+
+    /// Moves the Commitment into a CompressedRistretto point.
+    public fun commitment_into_point(c: Commitment): RistrettoPoint {
+        let Commitment { point } = c;
+        point
+    }
+
+    /// Moves the Commitment into a CompressedRistretto point.
+    public fun commitment_into_compressed_point(c: Commitment): CompressedRistretto {
+        point_compress(&c.point)
+    }
+
+    /// Returns the randomness base compatible with the Bulletproofs module.
```

Comment:
What does "compatible with the Bulletproofs module" mean here?

Reply:
A better comment would be:

> A Bulletproof attests that a value $v$ inside a Pedersen commitment $g^v h^r$ is sufficiently "small" (e.g., is 32-bits wide). Here, $h$ is referred to as the "randomness base" of the commitment scheme.
> Bulletproof has a default choice for $g$ and $h$ and this function simply returns that default $h$.

Reply:
Can we update this comment as per above?

Reply:
Done, with small edits